modules/assim.sequential/R/sda.enkf.R

```diff
@@ -129,176 +145,382 @@ sda.enkf <- function(settings,IC,prior,obs){
       
   ## start model run
   start.model.runs(settings,settings$database$bety$write)
-  
-
-  time = start.year:end.year
-  nt = length(time)
-  ens = list()
-  NPPm = rep(NA,nens)
+   
+  total.time = as.numeric(end.year):(as.numeric(end.year)+nrow(obs.mean))
```

Comment:
It should be start.year not end.year. Typo. For the tree ring dataset there is only one observation per year that's why the number of rows determines the years. I haven't thought about how to do more than one observation per year. We could add a column to the observations that has the year in it? This will affect later code that reads the observations too. This made me see an earlier problem on line 32. I guess we should just talk about this in person to make sure we're on the same page with time and observations.